packages/lib/services/interop/InteropService_Importer_Md.ts

```diff
@@ -59,52 +63,104 @@ export default class InteropService_Importer_Md extends InteropService_Importer_
 	 */
-	async importLocalFiles(filePath: string, md: string) {
+	async importLocalFiles(filePath: string, md: string, parentFolderId: string) {
 		let updated = md;
-		const fileLinks = unique(markdownUtils.extractFileUrls(md));
+		const markdownLinks = markdownUtils.extractFileUrls(md);
+		const htmlLinks = htmlUtils.extractFileUrls(md);
+		const fileLinks = unique(markdownLinks.concat(htmlLinks));
 		await Promise.all(fileLinks.map(async (encodedLink: string) => {
 			const link = decodeURI(encodedLink);
-			const attachmentPath = filename(`${dirname(filePath)}/${link}`, true);
-			const pathWithExtension = `${attachmentPath}.${fileExtension(link)}`;
+			// Handle anchor links appropriately
+			const trimmedLink = this.trimAnchorLink(link);
+			const attachmentPath = filename(`${dirname(filePath)}/${trimmedLink}`, true);
+			const pathWithExtension = `${attachmentPath}.${fileExtension(trimmedLink)}`;
 			const stat = await shim.fsDriver().stat(pathWithExtension);
 			const isDir = stat ? stat.isDirectory() : false;
 			if (stat && !isDir) {
-				const resource = await shim.createResourceFromPath(pathWithExtension);
-				// NOTE: use ](link) in case the link also appears elsewhere, such as in alt text
-				const linkPatternEscaped = pregQuote(`](${link})`);
-				const reg = new RegExp(linkPatternEscaped, 'g');
-				updated = updated.replace(reg, `](:/${resource.id})`);
+				const supportedFileExtension = this.metadata().fileExtensions;
+				const resolvedPath = shim.fsDriver().resolve(pathWithExtension);
+				let id: string = '';
+				// If the link looks like a note, then import it
+				if (supportedFileExtension.indexOf(fileExtension(trimmedLink).toLowerCase()) >= 0) {
+					// If the note hasn't been imported yet, do so now
+					if (!this.importedNotes[resolvedPath]) {
+						await this.importFile(resolvedPath, parentFolderId);
+					}
+
+					id = this.importedNotes[resolvedPath].id;
+				} else {
+					const resource = await shim.createResourceFromPath(pathWithExtension);
+					id = resource.id;
+				}
+
+				// The first is a normal link, the second is supports the <link> and [](<link with spaces>) syntax
+				// Only opening patterns are consider in order to cover all occurances
+				// We need to use the encoded link as well because some links (link's with spaces)
+				// will appear encoded in the source. Other links (unicode chars) will not
+				const linksToReplace = [this.trimAnchorLink(link), this.trimAnchorLink(encodedLink)];
+
+				for (let j = 0; j < linksToReplace.length; j++) {
+					const linkToReplace = pregQuote(linksToReplace[j]);
+
+					// Markdown links
+					const linkRegex = `(?<=\\]\\()\\<?${linkToReplace}\\>?(?=.*\\))`;
+					const reg = new RegExp(linkRegex, 'g');
+					updated = updated.replace(reg, `:/${id}`);
+
+					// HTML links
+					const htmlLinkRegex = `(?<=(?:src|href)=["'])${linkToReplace}(?=["'])`;
+					const htmlReg = new RegExp(htmlLinkRegex, 'g');
+					updated = updated.replace(htmlReg, `:/${id}`);
```

Comment:
And this to lib/htmlUtils? This is to avoid having too many subtly different regexes for the same thing scattered around the code. At least if they are all under the same file it's easier to manage.

Reply:
Good idea, I should have implemented it like that from the beginning.